packages/providers/src/provider.ts

```diff
@@ -205,28 +207,75 @@ export type ProviderCallParams = {
   utxoValidation?: boolean;
 };
 
+/**
+ * URL - Consensus Params mapping.
+ */
+type ConsensusParamsCache = Record<string, ConsensusParameters>;
+
 /**
  * A provider for connecting to a node
  */
 export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
+  consensusParamsCache: ConsensusParamsCache = {};
 
   /**
    * Constructor to initialize a Provider.
    *
    * @param url - GraphQL endpoint of the Fuel node
+   * @param consensusParams - Consensus parameters of the Fuel node
    * @param options - Additional options for the provider
+   * @hidden
    */
   constructor(
     /** GraphQL endpoint of the Fuel node */
     public url: string,
+    consensusParams: ConsensusParameters,
     public options: ProviderOptions = {}
   ) {
     this.operations = this.createOperations(url, options);
+    this.consensusParamsCache[url] = consensusParams;
     this.cache = options.cacheUtxo ? new MemoryCache(options.cacheUtxo) : undefined;
   }
 
+  /**
+   * Creates a new instance of the Provider class. This is the recommended way to initialize a Provider.
+   * @param url - GraphQL endpoint of the Fuel node
+   * @param options - Additional options for the provider
+   */
+  static async connect(url: string, options: ProviderOptions = {}) {
+    const consensusParameters = await this.getConsensusParamsWithoutInstance(url);
+    const provider = new Provider(url, consensusParameters, options);
+    return provider;
+  }
```

Comment:
we could apply this strategy to don't have breaking changes:

```suggestion
  static async create({ url, options }: { url: string, options?: ProviderOptions }) {
    const provider = new Provider(url, options);
    await provider.connect();
    return provider;
  }
```

on this way:
- use a more generic name `create` that can be used in other classes as well to maintain the pattern
- no breaking changes as constructor is not changed (everyone using `new Provider(url)`)
- use input object to pass params
- `updateUrl` function is not needed as `connect` should be centralizing the connection and caching the provider info when needed 
- no breaking changes in `connect` method as won't be changed to `static async`. 
  - `connect` now will receive a optional `url` param, this allow us to connect to current URL and also change to other URL
  - when connecting will get the url from:
```
  connect({ url }: { url: string }) {
    const urlToConnect = input?.url || this.url; 
    // logic to connect to URL, get consensus params if it's not cached, etc..
  }
```

ps. we can also adopt same pattern to use a `static async create` as a way to handle the async part that we don't wanna do in the constructor. other places in the code might need a similar refactor to avoid hardcoded consts

it's being done also in TransactionResponse -> https://github.com/FuelLabs/fuels-ts/pull/1206

Reply:
> I like this idea @LuizAsFight, but wouldn't this _still_ be a breaking change for most users? For example, anyone who has implemented any contract calls will need to update their code and add `await provider.create()` everywhere.

 yeah about this I'm ok with changing to remove `new Provider` and always use `Provider.create` if you prefer on that way

Reply:
keeping `connect` function as I mentioned in other comments instead of changing to `updateUrl` will also help with breaking changes

Reply:
@LuizAsFight Tysm for your detailed review of the PR and so many helpful suggestions mate! After careful consideration of all of them and some discussions with the team, I have incorporated some of them but have decided to keep some things as they were. The team has agreed that it is better to have this *loud* breaking change instead of keeping it _silent_ and leaving users with some code that might seem ok but actually needs some additions in order to work now. That itself would be a breaking change, and potentially even dangerous.